crates/cext/src/circuit.rs

```diff
@@ -560,31 +563,48 @@ pub unsafe extern "C" fn qk_circuit_get_instruction(
     let out_params = params_vec.as_mut_ptr();
     std::mem::forget(params_vec);
 
-    CInstruction {
-        name: CString::new(packed_inst.op.name()).unwrap().into_raw(),
-        num_qubits: qargs.len() as u32,
-        qubits: out_qargs,
-        num_clbits: cargs.len() as u32,
-        clbits: out_cargs,
-        num_params: params.len() as u32,
-        params: out_params,
+    unsafe {
+        *instruction = CInstruction {
+            name: CString::new(packed_inst.op.name()).unwrap().into_raw(),
+            num_qubits: qargs.len() as u32,
+            qubits: out_qargs,
+            num_clbits: cargs.len() as u32,
+            clbits: out_cargs,
+            num_params: params.len() as u32,
+            params: out_params,
+        }
     }
 }
 
 /// @ingroup QkCircuit
 /// Free a circuit instruction object.
 ///
+/// This function doesn't free the allocation for the provided ``QkInstruction`` pointer, it
+/// only free the internal allocations for the data contained in the instruction. You are
+/// responsible for allocating and freeing the actually allocation used to store a
+/// ``QkCircuitInstruction``.
+///
 /// @param inst The instruction to free.
 ///
+/// # Example
+///
+///     QkCircuitInstruction *inst = malloc(sizeof(QkCircuitInstruction));
+///     QkCircuit *qc = qk_circuit_new(100);
+///     qk_circuit_gate(qc, QkGate_H, *[0], *[]);
```

Comment:
Error: Wishful syntax 🙂

Reply:
More just sloppy copy and paste from someone's suggestions on #14006 

```suggestion
///     qk_circuit_gate(qc, QkGate_H, *[0], NULL);
```